goth/runner/exceptions.py

```diff
@@ -19,7 +19,7 @@ class KeyAlreadyExistsError(CommandError):
     """Specific duplicate key subclass of the CommandError."""
 
 
-class StopThreadException(Exception):
+class StopThreadException(BaseException):
```

Comment:
This makes `StopThreadException` less likely to be caught accidentally in a thread that it is intended to stop.